### PR TITLE
Feat/client metadata

### PR DESCRIPTION
**Pull request by adrukh**
Broker clients will now identify themselves with the broker client by passing their metadata in addition to their token.

The purpose of this change is to improve visibility into live broker clients for better troubleshooting etc.

### CODE REVIEW

File: lib/client/socket.js [hunk 1]
```diff
@@ -42,8 +42,9 @@ module.exports = ({ url, token, filters, config }) => {
     }
   });
   io.on('open', () => {
-    logger.info({ token, url }, 'identifying');
-    io.send('identify', token);
+    logger.info({ token, url }, 'identifying with broker server');
+    const clientData = { url, token, filters, config };
+    io.send('identify', JSON.stringify(clientData));
```
Comment (adrukh) [inline lib/client/socket.js]: I'm stringifying the client data to parse it on the server side. Should I just send the object as is, and read it on the other side? Didn't check if this is feasible.

Comment (gjvis) [reply lib/client/socket.js]: I'm pretty sure engine-io will handle the serialization for us, so shouldn't need to handle the JSON encoding manually

File: lib/server/socket.js [hunk 2]
```diff
@@ -34,13 +34,26 @@ module.exports = ({ server, filters, config }) => {
...
+      if (!token) {
+        logger.warn({ clientData }, 'client identified without a token');
+        return;
+      }
+
+      logger.info({ clientData, token }, 'client identified');
+
       const clientPool = connections.get(token) || [];
-      clientPool.unshift(socket);
+      clientPool.unshift({ socket, metadata: clientData.metadata });
```
Comment (adrukh) [inline lib/server/socket.js]: connections map now holds objects with socket and metadata

File: lib/client/index.js [hunk 3]
```diff
@@ -1,9 +1,10 @@
 const socket = require('./socket');
 const relay = require('../relay');
 const logger = require('../log');
+const version = require('../../package.json').version || 'local';
```
Comment (gjvis) [inline lib/client/index.js]: Can we wrap this up into `lib/version.js` please, then reference it appropriately

Comment (adrukh) [reply lib/client/index.js]: Will do

**Pull request merged**